File: ruby/0023.rb
def length(xa, ya, xb, yb)
  Math.sqrt((xa - xb)**2 + (ya - yb)**2)
end

n = gets.to_i
while gets
  xa, ya, ra, xb, yb, rb = $_.split(' ').map(&:to_f)
  l = length(xa, ya, xb, yb)
  if ra > l + rb
    out = 2
  elsif rb > l + ra
    out = -2
  elsif ra + rb < l
    out = 0
  else
    out = 1
  end
  puts out
end

File: python/0023.py
import math

def length(xa, ya, xb, yb):
  return math.sqrt((xa - xb)**2 + (ya - yb)**2)

n = int(raw_input())
for s in range(0, n):
  xa, ya, ra, xb, yb, rb = map(float, raw_input().split(' '))
  d = length(xa, ya, xb, yb)

  if ra > rb + d:
    print 2
  elif rb > ra + d:
    print -2
  elif d > ra + rb:
    print 0
  else:
    print 1